src/index.js

```diff
@@ -69,45 +78,45 @@ async function report(checks = {}, timeout = 10000) {
     return {
       statusCode: 200,
       headers: {
-        'Content-Type': 'application/xml',
+        'Content-Type': decorator.mime,
         'X-Version': version,
       },
-      body: [
-        '<pingdom_http_custom_check>',
-        '  <status>OK</status>',
-        `  <version>${version}</version>`,
-        `  <response_time>${Math.abs(Date.now() - start)}</response_time>`,
-        ...checkresults.map(({ key, response }) => `  <${key}>${Math.floor(response.timings.end)}</${key}>`),
-        '  <process>',
-        `    <activation>${process.env.__OW_ACTIVATION_ID}</activation>`,
-        '  </process>',
-        '</pingdom_http_custom_check>',
-      ].join('\n'),
+      body: decorator.body({
+        status: 'OK',
+        version,
+        response_time: Math.abs(Date.now() - start),
+        process: {
+          activation: process.env.__OW_ACTIVATION_ID,
+        },
+        ...checkresults.reduce((p, { key, response }) => {
+          // eslint-disable-next-line no-param-reassign
+          p[key] = Math.floor(response.timings.end);
+          return p;
+        }, {}),
+      }, decorator.name),
     };
   } catch (e) {
     const statusCode = (e.response ? e.response.statusCode : '') || 500;
     const body = (e.response ? e.response.body : '') || e.message;
     return {
       statusCode,
       headers: {
-        'Content-Type': 'application/xml',
+        'Content-Type': decorator.mime,
         'X-Version': version,
       },
-      body: [
-        '<pingdom_http_custom_check>',
-        '  <status>failed</status>',
-        `  <version>${version}</version>`,
-        `  <response_time>${Math.abs(Date.now() - start)}</response_time>`,
-        '  <error>',
-        `    <url>${e.options.uri}</url>`,
-        `    <statuscode>${statusCode}</statuscode>`,
-        `    <body><![CDATA[${body}]]></body>`,
-        '  </error>',
-        '  <process>',
-        `    <activation>${process.env.__OW_ACTIVATION_ID}</activation>`,
-        '  </process>',
-        '</pingdom_http_custom_check>',
-      ].join('\n'),
+      body: decorator.body({
+        status: 'failed',
+        version,
+        response_time: Math.abs(Date.now() - start),
+        error: {
+          url: e.options.uri,
+          statuscode: statusCode,
+          body: `<![CDATA[${body}]]>`,
```

Comment:
I don't think we should add XML CDATA inside the JSON.

Reply:
Yes.